homeassistant/components/overkiz/select.py

```diff
@@ -0,0 +1,119 @@
+"""Support for Overkiz select."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantOverkizData
+from .const import DOMAIN, IGNORED_OVERKIZ_DEVICES
+from .entity import OverkizDescriptiveEntity
+
+
+@dataclass
+class OverkizSelectDescriptionMixin:
+    """Define an entity description mixin for select entities."""
+
+    options: list[str]
+    select_option: Callable[[str, Callable], None]
+
+
+@dataclass
+class OverkizSelectDescription(SelectEntityDescription, OverkizSelectDescriptionMixin):
+    """Class to describe an Overkiz select entity."""
+
+
+SELECT_DESCRIPTIONS: list[OverkizSelectDescription] = [
+    OverkizSelectDescription(
+        key=OverkizState.CORE_OPEN_CLOSED_PEDESTRIAN,
+        name="Position",
+        icon="mdi:content-save-cog",
+        options=[
+            OverkizCommandParam.OPEN,
+            OverkizCommandParam.PEDESTRIAN,
+            OverkizCommandParam.CLOSED,
+        ],
+        select_option=lambda option, execute_command: execute_command(
+            {
+                OverkizCommandParam.CLOSED: OverkizCommand.CLOSE,
+                OverkizCommandParam.OPEN: OverkizCommand.OPEN,
+                OverkizCommandParam.PEDESTRIAN: OverkizCommand.SET_PEDESTRIAN_POSITION,
+            }[option]
```

Comment:
If the lambda is multiple lines, it is usually recommended to move it out of the `EntityDescription` and to simply reference the function name:
```suggestion
        select_option=_select_option_position
```

Reply:
I didn't like those multi-lines lambda's as well, however I am not sure if moving them out of the description will improve readability. If the lambda's are referenced and the originals are somewhere else in the file, you don't have the clear overview of what each entity description does?

Happy to move them if this is the best solution eventually :-).

Reply:
Done in ea79ed1.